src/manage.py

```
@@ -50,6 +54,61 @@
 glance = os_client_config.make_client("image", cloud=CONF.cloud)
 
 
+def process_local_image(glance, name, image, url):
+    logging.info("Create glance image '%s'" % name)
+    image_properties = {
+        'container_format': 'bare',
+        'disk_format': image['format'],
+        'min_disk': image.get('min_disk', 0),
+        'min_ram': image.get('min_ram', 0),
+        'name': name,
+        'tags': [CONF.tag],
+        'visibility': 'private'
+    }
+
+    file_name = url.split('/')[-1]
+    tmp_image = str(pathlib.Path.cwd() / file_name)
+    tmp_image_raw = tmp_image + '.raw'
+    logging.info("Downloading image for convertion and/or decompression: '%s'" % name)
+    with requests.get(url, stream=True) as response:
+        response.raise_for_status()
+        with open(tmp_image, 'wb') as out_file:
+            out_file.write(response.content)
```

This does not stream the content, `response.content` will always return all data as a bytes object (so all data will be cached in RAM). The correct solution seems to loop ovder `response.iter_content(chunk_size=None)`  and write the chunks <https://docs.python-requests.org/en/master/api/#requests.Response.iter_content>.

I am, however, not sure how how connection resets during the streaming of the contents are handled (if they just end the content iterator this would lead to silently truncated results).

---

I worked the `response.iter_content(chunk_size=None)` solution in. 

I would think connection reset would lead to exceptions like: `requests.exceptions.ChunkedEncodingError: ("Connection broken: ConnectionResetError(54, 'Connection reset by peer')", ConnectionResetError(54, 'Connection reset by peer'))`